lib/seattleflu/id3c/cli/command/etl/fhir.py

```diff
@@ -553,3 +558,29 @@ def observation_resource(device: str) -> Any:
             )
         )
     }
+
+
+def create_redcap_uri(record: REDCapRecord) -> str:
+    """
+    Create a JSON data URI representing the source of a given REDCap *record*.
+
+    Used in FHIR Encounter resources as the meta.source, which ultimately winds
+    up in ID3C's ``warehouse.encounter.details`` column.
+    """
+    data_scheme = 'data:application/json'
+
+    redcap_data = {
+        'url': 'https://redcap.iths.org',
+        'project_id': record.project.id,
+        'record_id': record.id
+    }
+
+    if record.get('redcap_event_name'):
+        redcap_data['event_name'] = record.get('redcap_event_name')
+
+    if record.get('redcap_repeat_instance'):
+        redcap_data['repeat_instance'] = record.get('redcap_repeat_instance')
```

Comment:
I'm slightly worried that `redcap_repeat_instance` could be `0` and we'd skip it here. Maybe REDCap doesn't allow that though?

I wonder about using `'redcap_repeat_instance' in record` instead (and ditto for event name for consistency)?

Reply:
REDCap doesn't allow repeat instance of 0, it returns this error when I try to import a record with instance 0:
```
Values being imported into a repeating instrument/event cannot have the 'redcap_repeat_instance' field with a value of '0'.
It must have a numeric value greater than '0'.
```

Reply:
I wanted to avoid ingesting empty strings. I believe currently the `enrollment_arm_1` event record has an empty string for `redcap_repeat_instance`

Reply:
Ah, great. Fine as-is then!